src/docs/timestamp-txn.dox

```diff
@@ -0,0 +1,155 @@
+/*! @page timestamp_txn_api Managing the transaction timestamp state
```

Comment:
@kommiharibabu , do you think we should also backport the full timestamp-txn document? The original PR only fixes up the comments in here. Let me know what you think.